### ISSUE TITLE
[prometheus-custom] prometheus organizes scrape target for every container in Pod with prometheus.deckhouse.io/port annotation

### ISSUE THREAD

**Issue by apolovov** [deckhouse/deckhouse#319]
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Version

v1.26

### Expected Behavior

There must be a single scrape target for metrics port, described with `prometheus.deckhouse.io/port` annotation. 

### Actual Behavior

There is a scrape target for every container in Pod with `prometheus.deckhouse.io/port` annotation. Even if the container hasn't containerPort.

### Additional Information

```
apiVersion: v1
kind: Service
metadata:
  name: pme
  labels:
    app: pme
    service: pme
spec:
  ports:
  - port: 2112
    name: http
  selector:
    app: pme
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: pme
  labels:
    app: pme
    version: v1
spec:
  replicas: 1
  selector:
    matchLabels:
      app: pme
      version: v1
  template:
    metadata:
      labels:
        app: pme
        version: v1
        prometheus.deckhouse.io/custom-target: processing
      annotations:
        prometheus.deckhouse.io/port: "2112"
        prometheus.deckhouse.io/path: '/metrics'
        prometheus.deckhouse.io/sample-limit: '15000'
    spec:
      containers:
      - name: pme
        image: driusha/prometheus-mock-exporter:latest
        resources:
          requests:
            cpu: "100m"
        imagePullPolicy: IfNotPresent #Always
        ports:
        - containerPort: 2112
      - name: dummy
        image: google/pause
```
And go to https://grafana.example.com/prometheus/targets

**Comment by nabokihms** [comment]
I checked the issue, and in my opinion, the most appropriate way to fix it is to drop the `container` label on endpoint discovery. Annotations only work on the scope of the pod, so it is impossible to have more than one scrape endpoint with the specified port.

Btw, thank you for the manifests for debugging. Much help!

**Comment by konstantin-axenov** [comment]
Resolved in 52c1550d9df166d57aed98b1357d2ae295c6bd7e
